spec/compiler/crystal/tools/implementations_spec.cr

```diff
@@ -368,4 +368,104 @@ describe "implementations" do
     Bar.bar
     )
   end
+
+  it "find class implementation" do
+    assert_implementations %(
+    ༓class Foo
+    end
+
+    F‸oo
+    )
+  end
+
+  it "find open class implementation" do
+    assert_implementations %(
+    ༓class Foo
+      def foo
+      end
+    end
+
+    ༓class Foo
+      def bar
+      end
+    end
+
+    F‸oo
+    )
+  end
+
+  it "find struct implementation" do
+    assert_implementations %(
+    ༓struct Foo
+    end
+
+    F‸oo
+    )
+  end
+
+  it "find module implementation" do
+    assert_implementations %(
+    ༓module Foo
+    end
+
+    F‸oo
+    )
+  end
+
+  it "find enum implementation" do
+    assert_implementations %(
+    ༓enum Foo
+      Foo
+    end
+
+    F‸oo
+    )
+  end
+
+  it "find enum value implementation" do
+    assert_implementations %(
+    enum Foo
+      ༓Foo
+    end
+
+    Foo::F‸oo
+    )
+  end
+
+  it "find class implementation via alias" do
+    assert_implementations %(
+    ༓class Foo
+    end
+
+    alias Bar = Foo
```

Comment:
Was there an explicit decision to not resolve to this line instead the Foo? What happens if the alias has a union?